metrics/src/agent.rs

```diff
@@ -0,0 +1,266 @@
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2021-Present Datadog, Inc.
+
+use crate::metrics::Metric;
+use crate::payload::construct_distribution_payload;
+
+use futures::lock::Mutex;
```

Comment:
maybe you don't need an external library and go with: https://doc.rust-lang.org/std/sync/struct.Mutex.html

Reply:
That doesn't work in an async context unfortunately, it's what we had originally.

Reply:
My understanding is that if we use a synchronous mutex, it'll block the entire thread, which would defeat the purpose of having the function be async in the first place.

Reply:
I switched it to use `tokio::sync::Mutex` as we're already using the `tokio` dependency, and removed the `futures` dependency.